Sources/BuildServerIntegration/SwiftToolchainResolver.swift

```diff
@@ -18,56 +18,86 @@ import TSCExtensions
+actor SwiftToolchainResolver {
   private struct CacheKey: Hashable {
     let compiler: URL
     let workingDirectory: URL?
   }
 
   private var cache: LRUCache<CacheKey, Result<URL?, Error>> = LRUCache(capacity: 100)
 
-  /// Check if `compiler` is a symlink to `swiftly`. If so, find the executable in the toolchain that swiftly resolves
-  /// to within the given working directory and return the URL of the corresponding compiler in that toolchain.
-  /// If `compiler` does not resolve to `swiftly`, return `nil`.
+  /// Check if `compiler` is a symlink to `swiftly` or in `/usr/bin` on macOS. If so, find the executable in the
+  /// toolchain that would be resolved to within the given working directory and return the URL of the corresponding
+  /// compiler in that toolchain. If `compiler` does not resolve to `swiftly` or `/usr/bin` on macOS, return `nil`.
   func resolve(compiler: URL, workingDirectory: URL?) async throws -> URL? {
-    let cacheKey = CacheKey(compiler: compiler, workingDirectory: workingDirectory)
+    let cacheKey: CacheKey = CacheKey(compiler: compiler, workingDirectory: workingDirectory)
     if let cached = cache[cacheKey] {
       return try cached.get()
     }
+
     let computed: Result<URL?, Error>
     do {
-      computed = .success(
-        try await resolveSwiftlyTrampolineImpl(compiler: compiler, workingDirectory: workingDirectory)
-      )
+      var resolved = try await resolveSwiftlyTrampoline(compiler: compiler, workingDirectory: workingDirectory)
+      if resolved == nil {
+        resolved = try await resolveXcrunTrampoline(compiler: compiler, workingDirectory: workingDirectory)
+      }
+
+      computed = .success(resolved)
     } catch {
       computed = .failure(error)
     }
+
     cache[cacheKey] = computed
     return try computed.get()
   }
 
-  private func resolveSwiftlyTrampolineImpl(compiler: URL, workingDirectory: URL?) async throws -> URL? {
+  private func resolveSwiftlyTrampoline(compiler: URL, workingDirectory: URL?) async throws -> URL? {
     let realpath = try compiler.realpath
     guard realpath.lastPathComponent == "swiftly" else {
       return nil
     }
+
     let swiftlyResult = try await Process.run(
       arguments: [realpath.filePath, "use", "-p"],
       workingDirectory: try AbsolutePath(validatingOrNil: workingDirectory?.filePath)
     )
     let swiftlyToolchain = URL(
       fileURLWithPath: try swiftlyResult.utf8Output().trimmingCharacters(in: .whitespacesAndNewlines)
     )
+
     let resolvedCompiler = swiftlyToolchain.appending(components: "usr", "bin", compiler.lastPathComponent)
     if FileManager.default.fileExists(at: resolvedCompiler) {
       return resolvedCompiler
     }
     return nil
   }
 
+  private func resolveXcrunTrampoline(compiler: URL, workingDirectory: URL?) async throws -> URL? {
+    #if !os(macOS)
+    return nil
+    #endif
+
+    guard compiler.deletingLastPathComponent() == URL(fileURLWithPath: "/usr/bin") else {
```

Comment:
We can avoid a trip to the file system here to check if `/usr/bin` is a directory
```suggestion
    guard compiler.deletingLastPathComponent() == URL(filePath: "/usr/bin/") else {
```